src/read.c

```diff
@@ -1896,9 +1888,41 @@ static avifBool avifParseCodecConfigurationBoxProperty(avifProperty * prop,
     config->chromaSubsamplingX = (rawFlags >> 3) & 0x1;      // unsigned int (1) chroma_subsampling_x;
     config->chromaSubsamplingY = (rawFlags >> 2) & 0x1;      // unsigned int (1) chroma_subsampling_y;
     config->chromaSamplePosition = (rawFlags >> 0) & 0x3;    // unsigned int (2) chroma_sample_position;
+
+    // unsigned int (3) reserved = 0;
+    // unsigned int (1) initial_presentation_delay_present;
+    // if (initial_presentation_delay_present) {
+    //   unsigned int (4) initial_presentation_delay_minus_one;
+    // } else {
+    //   unsigned int (4) reserved = 0;
+    // }
+    AVIF_CHECK(avifROStreamSkip(s, 1));
+
+    // According to section 2.2.1. of AV1 Image File Format specification v1.1.0:
+    //   "- Sequence Header OBUs should not be present in the AV1CodecConfigurationBox."
+    //   "- If a Sequence Header OBU is present in the AV1CodecConfigurationBox,
+    //      it shall match the Sequence Header OBU in the AV1 Image Item Data."
+    //   "- Metadata OBUs, if present, shall match the values given in other item properties,
+    //      such as the PixelInformationProperty or ColourInformationBox."
+    // See https://aomediacodec.github.io/av1-avif/v1.1.0.html#av1-configuration-item-property.
+    // For simplicity, the constraints above are not enforced.
+    // The following is skipped by avifParseItemPropertyContainerBox().
+    // unsigned int (8) configOBUs[];
     return AVIF_TRUE;
 }
 
+static avifBool avifParseCodecConfigurationBoxProperty(avifProperty * prop,
+                                                       const uint8_t * raw,
+                                                       size_t rawLen,
+                                                       const char * configPropName,
+                                                       avifDiagnostics * diag)
+{
+    char diagContext[] = "Box[....]";
+    snprintf(diagContext, sizeof(diagContext), "Box[%.4s]", configPropName); // "Box[av1C]" or "Box[av2C]"
+    BEGIN_STREAM(s, raw, rawLen, diag, diagContext);
+    return avifParseCodecConfiguration(&s, &prop->u.av1C, configPropName, diag);
```

Comment:
As written above, because this is partly to simplify https://github.com/AOMediaCodec/libavif/pull/1432. Also it matches the [section 2.3.3](https://aomediacodec.github.io/av1-isobmff/v1.2.0.html#av1codecconfigurationbox-syntax) which is also split into two functions.

Reply:
The two functions here do not seem to match the two functions in [section 2.3.3](https://aomediacodec.github.io/av1-isobmff/v1.2.0.html#av1codecconfigurationbox-syntax).

The only thing the outer function (`avifParseCodecConfigurationBoxProperty`) does is to set up the `avifROStream s` variable, which is only used by the inner function (`avifParseCodecConfiguration`). So the outer function does essentially no parsing.

In contrast, the outer type in [section 2.3.3](https://aomediacodec.github.io/av1-isobmff/v1.2.0.html#av1codecconfigurationbox-syntax) is:

```
class AV1CodecConfigurationBox extends Box('av1C'){
  AV1CodecConfigurationRecord av1Config;
}
```

So to match [section 2.3.3](https://aomediacodec.github.io/av1-isobmff/v1.2.0.html#av1codecconfigurationbox-syntax), I would expect the outer function (`avifParseCodecConfigurationBoxProperty`) to parse the `Box('av1C')` header, but it doesn't do that.

In any case, this is fine if this helps simplify https://github.com/AOMediaCodec/libavif/pull/1432.